kubernetes/kubernetes.go

```diff
@@ -492,13 +526,61 @@ func (ka KubeAgentConfig) sendMetricsToCustomS3(metricSample *os.File) {
 	}
 }
 
-// generateSampleKey creates a key (location) for s3 to upload the sample to. Example of s3 location format
+func (ka KubeAgentConfig) sendMetricsToCustomBlob(metricSample *os.File) {
+	os.Setenv("AZURE_TENANT_ID", ka.CustomAzureTenantID)
```

Comment:
is setting the environment variables the best practice here?
Can they not just be injected into the cred/client?

Reply:
couldn't find a good example not setting the envs will double check w/ John tomorrow about this.

Reply:
Updated to now use: https://github.com/Azure/azure-sdk-for-go/blob/sdk/azidentity/v1.8.0/sdk/azidentity/client_secret_credential.go#L47 no longer setting the envs that way.